### PR TITLE
add RetroShare statistics

### PR DESCRIPTION
**Pull request by sehraf**
this needs https://github.com/RetroShare/RetroShare/pull/488 to be merged which should happen in the near future

### CODE REVIEW

**Comment by dtschmitz** [comment]
Nice work.

**Comment by ktsaou** [comment]
nice! thanks!

**Comment by ktsaou** [comment]
@paulfantom please review and merge this if it is ok with you.

File: conf.d/python.d/retroshare.conf [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+#
+# job_name:
+#     name: myname     # the JOB's name as it will appear at the
+#                      # dashboard (by default is the job_name)
+#                      # JOBs sharing a name are mutually exclusive
+#     update_every: 1  # the JOB's data collection frequency
+#     priority: 60000  # the JOB's order on the dashboard
+#     retries: 5       # the JOB's number of restoration attempts
+#
+# Additionally to the above, example also supports the following:
```
Comment (ktsaou) [inline conf.d/python.d/retroshare.conf]: minor: this is not the `example` module.

**Comment by ktsaou** [comment]
suggestion: you could also add a few alarms to this PR, to make sure you receive notifications if RetroShare is facing difficulties...

Check the `conf.d/health.d` directory for examples.

**Comment by sehraf** [comment]
I'll fix the conf file and have a look at the alarms. 
Thx for the suggestion

**Comment by ktsaou** [comment]
nice!

another suggestion: you can edit `web/index.html` to set a better menu item, icon (from font-awesome) and probably chart information to the RetroShare charts.

This is the place you should look: https://github.com/firehol/netdata/blob/ae273fdf4bcd99bab8953d0fd233c5381bd5374c/web/index.html#L1494-L1694

Below the highlighted section you can add information per sub-section (submenu) and per chart.

**Comment by sehraf** [comment]
Ahh thx for pointing this out. I was looking fr this but didn't find it. It will definitely improve the chats!

Are you ok with not merging until i've modified `web/index.html` (might take me a few days) or should i send a second pull request later?

**Comment by ktsaou** [comment]
it is ok we will wait. We also depend on the RetroShare PR (it would be funny to add this to netdata before RetroShare merges the PR to expose this information).

**Comment by ktsaou** [comment]
I added the `wontmerge` flag to prevent up from accidentally merging it.
When you are ready, post here.

**Comment by ktsaou** [comment]
bad quotation. I changed it to `enhancement` and `in progress` :)

**Comment by sehraf** [comment]
I'd like to use the easypiechart to display the number of connected friends compared to all friends but it seems to me that the easypiechart needs a constant max value (so i can't use the number off all friends for max value since it's variable). 
Is there another way to do that? Can i calculate the percentage of connected peers in `web/indec.html`?

**Comment by ktsaou** [comment]
> it seems to me that the easypiechart needs a constant max value

no it does not. All charts in netdata calculate the max value themselves. Just don't set a max value.
Check how the `mainheads` in `index.html` are rendered.

**Comment by sehraf** [comment]
but then the max value is the highest number of connected friends seen so far and not the number of all friends

**Comment by ktsaou** [comment]
> but then the max value is the highest number of connected friends seen so far and not the number of all friends

you are right. This is however the standard so far for netdata....

**Comment by sehraf** [comment]
i'd say i'm done. Are there any further suggestions or improvements?

**Comment by ktsaou** [comment]
Can you post a few screenshots from the dashboard?
Just paste images at github.

**Comment by sehraf** [comment]
This is how it looks:
netdata-rs-1
netdata-rs-2

**Comment by ktsaou** [comment]
looks good! nice work!
So, we can wait for retroshare to merge their PR.

**Comment by sehraf** [comment]
this already happened 😸 
https://github.com/RetroShare/RetroShare/commit/798ffd85ab0d4e95bf27df042e0a981eec3fc6d2

**Comment by ktsaou** [comment]
merged! thanks!

**Comment by ktsaou** [comment]
@paulfantom I merged this, but now I realized you haven't reviewed this. Could you please check this module?

**Pull request merged**